Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -181,6 +181,16 @@ class BrowserViewController: UIViewController {
         }, completion: { _ in
             self.scrollController.setMinimumZoom()
         })
+        
+        coordinator.animate(alongsideTransition: { context in
+            if let webView = self.tabManager.selectedTab?.webView {
+                webView.frame = self.view.bounds
```

Comment:
I find it interesting because I dug a little into it and couldn't even find a point where the web view's frame is updated based on the new `webViewContainer`'s frame (which is constrained) other than on `didCreateWebView`…  So I'm not sure how this ever worked in landscape mode 😅

I think the idea of adding the update to the webView's frame is fine, but we may want to consider doing it in `viewDidLayoutSubviews` so its always based on `self.webViewContainer.bounds` and not temporarily based on the view's bounds.

Reply:
Ah, found where its set:

https://github.com/brave/brave-ios/blob/4b540b4d209ee5411ded1d491603fac2e3c3c07a/Client/Frontend/Browser/BrowserViewController.swift#L2019-L2021

If it's constrained, is there a reason why we need to animate it alongside the coordinator to the view's bounds?

Reply:
@kylehickinson Yeah I have ZERO idea why that is. It's not the WebView's fault. It's Google's fault for their weird JS on Youtube only (so far).

Who knows what other sites have weird meta tags or JS to modify the video size based on view-port.. The problem I found is that Youtube does it "early".. IE: they do it the same time as the rotation instead of on the next layout cycle. Vimeo and other sites do it correctly.

Not sure how Safari handles it but it works for Apple lol.

Actually, if we scroll after the bug has happened, it fixes itself which is why I originally thought programatically scrolling would fix it but it doesn't. I can try adding the layoutSubviews too (I tried it by itself, it doesn't work at all without this animateAlongSide fix).

Reply:
Its just that we technically are constraining it, so setting its frame is actually going to get override anyways at next layout cycle. Not sure if that can happen during the willTransition methods though. Perhaps we can try to force relayout with `setNeedsLayout`/`layoutIfNeeded` instead?

Reply:
Originally I programatically injected `<meta name="viewport" content="width=device-width, initial-scale=1">` but youtube already has that.. so I figured I'd change `width=device-width` to the actual size and it works but not every time until the viewport change gets noticed (requires window.scrollBy(1, 1)`, etc..

I ended up making a separate test application with WKWebView constrained to view and rotating.. same bug happens and the same fix fixes it. :S

Reply:
But is it because `webViewContainer` doesn't have the right frame? I'm not sure why we have to set it to `view.bounds`

Reply:
Fixed. I gave up on the solution of modifying the frame. The webViewContainer had the right frame. The webView's `WKContentView (private)` did not.

I changed the solution to use Javascript. It should make us all feel safer I hope.